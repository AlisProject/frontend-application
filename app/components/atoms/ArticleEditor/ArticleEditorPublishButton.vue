<template>
  <span class="nav-link post-article" :class="{ disable: !publishable }" @click="onClick">
    公開する
  </span>
</template>

<script>
export default {
  props: {
    publishable: Boolean,
    onClick: Function
  }
}
</script>

<style lang="scss" scoped>
.post-article {
  color: #858dda;
  cursor: pointer;
  user-select: none;
  display: inline-block;

  &.disable {
    cursor: not-allowed;
  }
}
</style>
