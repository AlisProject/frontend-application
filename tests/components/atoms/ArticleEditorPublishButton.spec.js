import { mount } from '@vue/test-utils'
import ArticleEditorPublishButton from '@/components/atoms/ArticleEditor/ArticleEditorPublishButton.vue'

describe('ArticleEditorPublishButton', () => {
  test('default snapshot ', () => {
    const wrapper = mount(ArticleEditorPublishButton)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('publishable snapshot ', () => {
    const wrapper = mount(ArticleEditorPublishButton, {
      propsData: {
        publishable: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
