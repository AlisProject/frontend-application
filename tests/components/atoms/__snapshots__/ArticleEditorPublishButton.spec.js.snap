// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleEditorPublishButton default snapshot  1`] = `
<span
  class="nav-link post-article disable"
>
  
  公開する

</span>
`;

exports[`ArticleEditorPublishButton publishable snapshot  1`] = `
<span
  class="nav-link post-article"
>
  
  公開する

</span>
`;
